pkg/scheduler/framework/v1alpha1/cycle_state.go

```diff
@@ -53,6 +55,22 @@ func NewCycleState() *CycleState {
 	}
 }
 
+// ShouldRecordFrameworkMetrics returns whether framework metrics should be recorded.
+func (c *CycleState) ShouldRecordFrameworkMetrics() bool {
+	if c == nil {
+		return false
+	}
+	return c.recordFrameworkMetrics
+}
+
+// RecordFrameworkMetrics sets recordFrameworkMetrics to the given value.
+func (c *CycleState) SetRecordFrameworkMetrics(flag bool) {
+	if c == nil {
+		return
+	}
+	c.recordFrameworkMetrics = flag
+}
+
```

Comment:
we can do this: 

```
func (c *CycleState) SetRecordFrameworkMetrics() {
	if c == nil {
		return
	}
        const sampleRate = 10  // 10% sample rate
        c.recordFrameworkMetrics = false
        if rand.Intn(100) < sampleRate {
          c.recordFrameworkMetrics = true
      }
}
```

```rand.Intn``` is in  ```"math/rand"```

Reply:
or is that going to make it more difficult to test?

Reply:
I initially considered rand but thought it was too expensive (I was using it for every plugin). But now since we now only need to sample once per schedule cycle, we can use rand.